src/renderer/components/+workloads-pods/pod-container-env/view.test.tsx

```diff
@@ -0,0 +1,277 @@
+/**
+ * Copyright (c) OpenLens Authors. All rights reserved.
+ * Licensed under MIT License. See LICENSE in root directory for more information.
+ */
+
+import React from "react";
+import type { ConfigMapStore } from "../../+config-maps/store";
+import configMapStoreInjectable from "../../+config-maps/store.injectable";
+import type { SecretStore } from "../../+config-secrets/store";
+import secretStoreInjectable from "../../+config-secrets/store.injectable";
+import type { PodContainer } from "../../../../common/k8s-api/endpoints";
+import { Secret, ConfigMap, Pod, SecretType } from "../../../../common/k8s-api/endpoints";
+import { getDiForUnitTesting } from "../../../getDiForUnitTesting";
+import type { DiRender } from "../../test-utils/renderFor";
+import { renderFor } from "../../test-utils/renderFor";
+import { ContainerEnvironment } from "./view";
+
+describe("<ContainerEnv />", () => {
+  let render: DiRender;
+  let secretStore: jest.Mocked<Pick<SecretStore, "load" | "getByName">>;
+  let configMapStore: jest.Mocked<Pick<ConfigMapStore, "load" | "getByName">>;
+
+  beforeEach(() => {
+    const di = getDiForUnitTesting({ doGeneralOverrides: true });
+
+    secretStore = ({
+      load: jest.fn().mockImplementation(async () => {
```

Comment:
nit: `jest.fn().mockImplementation(...)` is same as `jest.fn(...)`

Improvement for the future: the promise should be controlled, we should test how component looks when promise has no resolved yet. That's where `asyncFn()` comes in.

Reply:
That is incorrect (which is why I did it this way), the types flow differently if you first bind `jest.fn()` and then bind the implementation.